Hmm, 

> Caused by: org.hibernate.exception.GenericJDBCException: Error advancing (next) ResultSet position [IJ031040: Connection is not associated with a managed connection: org.jboss.jca.adapters.jdbc.jdk8.WrappedConnectionJDK8@1313c7dd] [n/a]
>   at org.hibernate@6.2.0.Final//org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:61)
>   at org.hibernate@6.2.0.Final//org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
>   at org.hibernate@6.2.0.Final//org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:94)
>   ... 170 more
> Caused by: java.sql.SQLException: IJ031040: Connection is not associated with a managed connection: org.jboss.jca.adapters.jdbc.jdk8.WrappedConnectionJDK8@1313c7dd
>   at org.jboss.ironjacamar.jdbcadapters@3.0.0.Final//org.jboss.jca.adapters.jdbc.WrappedConnection.lock(WrappedConnection.java:173)
>   at org.jboss.ironjacamar.jdbcadapters@3.0.0.Final//org.jboss.jca.adapters.jdbc.WrappedStatement.lock(WrappedStatement.java:138)
>   at org.jboss.ironjacamar.jdbcadapters@3.0.0.Final//org.jboss.jca.adapters.jdbc.WrappedResultSet.lock(WrappedResultSet.java:6022)
>   at org.jboss.ironjacamar.jdbcadapters@3.0.0.Final//org.jboss.jca.adapters.jdbc.WrappedResultSet.next(WrappedResultSet.java:2679)
>   at org.hibernate@6.2.0.Final//org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.advanceNext(JdbcValuesResultSetImpl.java:204)
>

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java
@@ -42,6 +42,7 @@ import org.jboss.shrinkwrap.api.Archive;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
@@ -124,6 +125,7 @@ public class SessionFactoryTestCase {
 
     // Test that a Persistence unit can be injected into a Hibernate Session factory
     @Test
+    @Ignore // WFLY-17830: renable when upgrading to Hibernate ORM 6.2.1.Final which should have the https://github.com/hibernate/hibernate-orm/pull/6350 change included
     public void testInjectPUIntoHibernateSessionFactory() throws Exception {
 
         SFSBHibernateSessionFactory sfsbHibernateSessionFactory =

